### DISCUSSION TITLE
lua-language-server exits on start

### DISCUSSION THREAD

**Discussion by Masterkraft0r** [helix-editor/helix#11472] [Troubleshooting] [answered]
### Summary

Hi all!

I'm trying to use the lua-language-server with helix and as soon as i open a lua file the line at the bottom says "Language server exited".

```
$ lua-language-server --version
3.10.3
```

```
$ hx --health lua
Configured language servers:
  ✓ lua-language-server: /home/bbugl/.asdf/shims/lua-language-server
Configured debug adapter: None
Configured formatter: None
Highlight queries: ✓
Textobject queries: ✓
Indent queries: ✓
```

### Reproduction Steps

1. Install lua-language-server via the tar or asdf, both are the same for me
2. Run `hx file.lua`
3. Read "Language server exited." at the bottom

### Helix log

<details><summary>~/.cache/helix/helix.log</summary>
```
2024-08-12T15:03:10.972 helix_lsp::transport [ERROR] Exiting lua-language-server after unexpected error: Parse(Error("data did not match any variant of untagged enum ServerMessage", line: 0, column: 0))
2024-08-12T15:03:10.972 helix_lsp [ERROR] failed to initialize language server: server closed the stream
2024-08-12T15:04:08.654 helix_lsp::transport [ERROR] lua-language-server err: <- StreamClosed
```
</details>

### Platform

Pop!_OS 22.04

### Terminal Emulator

alacritty

### Installation Method

apt via the ppa

### Helix Version

helix 24.7 (079f5442)

**Comment by the-mikedavis** [comment]
I can reproduce this. It looks like this is caused by https://github.com/LuaLS/lua-language-server/commit/40fd1d7c4c9bed4842acfdfe4e1cbb0307bab941 - the server sends on startup:

```
2024-08-12T14:20:50.324 helix_lsp::transport [INFO] lua-language-server <- {"jsonrpc":"2.0","method":"$/hello","params":"world"}
```

but this is invalid JSONRPC. Section 4.2 https://www.jsonrpc.org/specification requires `params` to be either an array or an object and we close the server because it sent an invalid JSONRPC notification. This should be reported upstream to lua-language-server

**Comment by yavorski** [comment, marked as answer]
Reported upstream https://github.com/LuaLS/lua-language-server/issues/2797

**Comment by Masterkraft0r** [comment]
Thank you for reporting this to the proper channels!
